### ISSUE TITLE
Custom line painting logic

### ISSUE THREAD

**Issue by subzero911** [imaNNeoFighT/fl_chart#523]
![image](https://user-images.githubusercontent.com/12999702/102902680-5e16fe80-4480-11eb-9888-9043c37d1719.png)

I need to paint the same line.
The `colors` property currently only allows to set the linear gradient over the whole line length. But here is a bit more complicated.
When the line is in the "low" zone, it should be blue.
When the line is in the "optimal" zone, it should be green.
When the line is in the "high" zone, it should be red.
Basically, I need **a vertical gradient,** not a horizontal.
How to implement it?

**Comment by subzero911** [comment]
Basically, I need a vertical gradient, not a horizontal.

**Comment by imaNNeoFighT** [comment]
I think we already have this feature.

Try using:
```
LineChartBarData(
  colors: [Colors.red, Colors.green, Colors.blue],
  gradientFrom: Offset(0, 0),
  gradientTo: Offset(0, 1),
),
```

![image](https://user-images.githubusercontent.com/7009300/103106249-f5c74900-4648-11eb-933c-8c3ae2c01609.png)

**Comment by imaNNeoFighT** [comment]
Feel free to close this issue if you are okay with my example.

Thanks!

**Comment by subzero911** [comment]
Thank you for your reply. We will test it tomorrow.

**Comment by subzero911** [comment]
Thanks, it works well.
